packages/zowe-explorer/src/job/ZosJobsProvider.ts

```diff
@@ -1158,6 +1160,66 @@ export class ZosJobsProvider extends ZoweTreeProvider implements IZoweTree<IZowe
             this.nodeDataChanged(session);
         }
     }
+
+    /**
+     * Updates or resets the filter for a given job.
+     * @param job The job whose filter should be updated/reset
+     * @param newFilter Either a valid `JobFilter` object, or `null` to reset the filter
+     * @param isSession Whether the node is a session
+     */
+    public updateFilterForJob(job: IZoweJobTreeNode, newFilter: string | null, isSession: boolean): void {
+        job.filter = newFilter;
+        job.description = newFilter ? localize("filter.description", "Filter: {0}", newFilter) : null;
+        this.nodeDataChanged(job);
+        if (newFilter === null) {
+            job["children"] = job["actualJobs"];
+            TreeProviders.job.refresh();
+        }
+    }
+
+    public async filterJobsDialog(job: IZoweJobTreeNode): Promise<vscode.InputBox> {
+        if (job.collapsibleState === vscode.TreeItemCollapsibleState.Collapsed) {
+            Gui.infoMessage(localize("filterJobs.message", "Use the search button to display jobs"));
+            return;
+        }
+        const selection = await Gui.showQuickPick(JOB_FILTER_OPTS, {
+            placeHolder: localize("filterJobs.placeholder", "Set a filter..."),
+        });
+        const filterMethod = JOB_FILTER_OPTS.indexOf(selection);
+
+        const isSession = contextually.isSession(job);
+        const userDismissed = filterMethod < 0;
+        if (userDismissed || selection === "$(clear-all) Clear filter for profile") {
+            if (selection === "$(clear-all) Clear filter for profile") {
```

Comment:
Although I'm not aware of any ongoing localizations for Zowe Explorer, can you please update this check so it compares the selection against the localized string?

```suggestion
        const clearFilterOpt = localize("filter.clearForProfile", "$(clear-all) Clear filter for profile");
        if (userDismissed || selection === clearFilterOpt) {
            if (selection === clearFilterOpt) {
```

This ensures that if/when localizations are created for a different language, this check will keep the same behavior regardless of language.